core/src/main/java/com/google/cloud/sql/core/DnsResolver.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.sql.core;
+
+import java.util.Collection;
+import javax.naming.NameNotFoundException;
+
+interface DnsResolver {
+  Collection<String> resolveTxt(String domainName) throws NameNotFoundException;
```

Comment:
we are using `dnsName` in other places, is `domainName` the same thing or is it a different meaning? If they are one and the same we should consolidate usage to one naming pattern

Reply:
Go connector uses `DomainName` as the configuration field. 

I considered `fqdn`, `dnsName`, `domainName`, `host`, and `hostname`. FQDN or FullyQualifiedDomainName is too wordy, and dnsName is an expands to "domain name system name", which is self-redundant. I was tempted `host`, or `hostname` because that is commonly what it is called as part of the a URL [RFC 1738](https://datatracker.ietf.org/doc/html/rfc1738#section-3.1). I decided against it because I wanted to be specific that it needs to contain the domain name. 

Go connector uses `DNSInstanceConnectionNameResolver` for the resolver class name, and `DNSResolver` for the singleton instance. This refers to the fact that it uses the DNS system to resolve the `DomainName` into an instance connection name. In retrospect, this felt too wordy, so the class name is `DNSResolver` in java.